lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -747,14 +747,18 @@ namespace Js
         return !!pActiveFuncs->TestAndSet(this->GetFunctionNumber());
     }
 
-    void FunctionBody::UpdateActiveFunctionsForOneDataSet(ActiveFunctionSet *pActiveFuncs, FunctionCodeGenRuntimeData **dataSet, uint count) const
+    void FunctionBody::UpdateActiveFunctionsForOneDataSet(ActiveFunctionSet *pActiveFuncs, FunctionCodeGenRuntimeData *parentData, FunctionCodeGenRuntimeData **dataSet, uint count) const
     {
         FunctionCodeGenRuntimeData *inlineeData;
         for (uint i = 0; i < count; i++)
         {
             for (inlineeData = dataSet[i]; inlineeData; inlineeData = inlineeData->GetNext())
             {
-                inlineeData->GetFunctionBody()->UpdateActiveFunctionSet(pActiveFuncs, inlineeData);
+                // inlineeData == parentData indicates a cycle in the structure. We've already processed parentData, so don't descend.
+                if (inlineeData != parentData)
```

Comment:
I have crash dumps, but I don't have a console repro that actually fails when I run it locally.